packages/valory/agents/hello_world/tests/test_hello_world.py

```diff
@@ -53,18 +55,113 @@
 )
 
 
-@pytest.mark.parametrize("nb_nodes", (4,))
-@pytest.mark.usefixtures(
-    "flask_tendermint", "tendermint_port", "abci_host", "abci_port"
-)
-class TestHelloWorldABCIFourAgents(
+# normal execution
+class BaseHelloWorldABCITest(
     BaseTestEnd2EndExecution,
 ):
-    """Test the ABCI hello_world skill with four agents."""
+    """Test the hello_world_abci skill with four agents."""
 
     agent_package = "valory/hello_world:0.1.0"
     skill_package = "valory/hello_world_abci:0.1.0"
     wait_to_finish = 160
     happy_path = HAPPY_PATH
-    strict_check_strings = STRICT_CHECK_STRINGS
+    strict_check_strings: Tuple[str, ...] = STRICT_CHECK_STRINGS
     package_registry_src_rel = Path(__file__).parent.parent.parent.parent.parent
+
+
+@pytest.mark.usefixtures(
+    "flask_tendermint", "tendermint_port", "abci_host", "abci_port"
+)
+@pytest.mark.parametrize("nb_nodes", (1,))
+class TestHelloWorldABCISingleAgent(BaseHelloWorldABCITest):
+    """Test the hello_world_abci skill with only one agent."""
+
+
+@pytest.mark.usefixtures(
+    "flask_tendermint", "tendermint_port", "abci_host", "abci_port"
+)
+@pytest.mark.parametrize("nb_nodes", (2,))
+class TestHelloWorldABCITwoAgents(BaseHelloWorldABCITest):
+    """Test the hello_world_abci skill with two agents."""
+
+
+@pytest.mark.usefixtures(
+    "flask_tendermint", "tendermint_port", "abci_host", "abci_port"
+)
+@pytest.mark.parametrize("nb_nodes", (4,))
+class TestHelloWorldABCIFourAgents(BaseHelloWorldABCITest):
+    """Test the hello_world_abci skill with four agents."""
+
+
+# catchup test
+class BaseHelloWorldABCITestCatchup(BaseHelloWorldABCITest):
+    """Test the hello_world_abci skill with catch up behaviour."""
+
+    stop_string = "register"
+    restart_after = 10
+    n_terminal = 1
+
+
+# four behaviours and different stages of termination and restart
+@pytest.mark.parametrize("nb_nodes", (4,))
+class TestHelloWorldABCIFourAgentsCatchupOnRegister(BaseHelloWorldABCITestCatchup):
+    """Test hello_world_abci skill with four agents; one restarting on `register`."""
+
+    stop_string = "register"
+
+
+@pytest.mark.parametrize("nb_nodes", (4,))
+class TestHelloWorldABCIFourAgentsCatchupRetrieveRandomness(
+    BaseHelloWorldABCITestCatchup
+):
+    """Test hello_world_abci skill with four agents; one restarting on `collect_randomness`."""
+
+    stop_string = "collect_randomness"
+
+
+@pytest.mark.parametrize("nb_nodes", (4,))
+class TestHelloWorldABCIFourAgentsCatchupSelectKeeper(BaseHelloWorldABCITestCatchup):
+    """Test hello_world_abci skill with four agents; one restarting on `select_keeper`."""
+
+    stop_string = "select_keeper"
+
+
+# multiple agents terminating and restarting
+@pytest.mark.parametrize("nb_nodes", (4,))
+class TestHelloWorldABCIFourAgentsTwoAgentRestarting(BaseHelloWorldABCITestCatchup):
+    """Test the hello_world_abci skill with four agents and two restarting."""
+
+    n_terminal = 2
+
+
+@pytest.mark.skip(reason="not working atm")
```

Comment:
??

Reply:
O_O

Reply:
Why all these tests - was it like this for simple abci?

Reply:
Yes I see it was

Reply:
@DavidMinarsch why are these tests skipped? - 9511c4b93ca46a7a5f5a36459f2afdd1a18328c7

Reply:
As far as I can see they did not work fine and were ignored before